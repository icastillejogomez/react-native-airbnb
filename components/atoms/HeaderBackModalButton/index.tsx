import { iconsSources } from '@/assets/icons'
import { usePalette } from '@/theme'
import { Image } from 'expo-image'
import { forwardRef, memo, useMemo } from 'react'
import { StyleSheet, TouchableOpacity } from 'react-native'

export type HeaderBackModalButtonProps = {
  onPress?: () => void
  tintColor?: string
  canGoBack?: boolean
}

const HeaderBackModalButton = memo(
  forwardRef<TouchableOpacity, HeaderBackModalButtonProps>((props, ref) => {
    const palette = usePalette()
    const tintColor = useMemo(() => props.tintColor || palette.icon.primary, [props.tintColor, palette])

    if (!props.canGoBack) {
      return null
    }

    return (
      <TouchableOpacity onPress={props.onPress} activeOpacity={0.8} style={styles.container} ref={ref}>
        <Image source={iconsSources.chevronLeft} style={{ width: 24, height: 24 }} tintColor={tintColor} />
      </TouchableOpacity>
    )
  }),
)
HeaderBackModalButton.displayName = 'HeaderBackModalButton'

export { HeaderBackModalButton }

const styles = StyleSheet.create({
  container: {
    // borderWidth: 1,
    // borderColor: 'red',
  },
})
