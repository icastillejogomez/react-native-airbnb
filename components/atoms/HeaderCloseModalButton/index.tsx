import { iconsSources } from '@/assets/icons'
import { usePalette } from '@/theme'
import { Image } from 'expo-image'
import { forwardRef, memo, useMemo } from 'react'
import { StyleSheet, TouchableOpacity } from 'react-native'

export type HeaderCloseModalButtonProps = {
  onPress?: () => void
  tintColor?: string
  canGoBack?: boolean
}

const HeaderCloseModalButton = memo(
  forwardRef<TouchableOpacity, HeaderCloseModalButtonProps>((props, ref) => {
    const palette = usePalette()
    const tintColor = useMemo(() => props.tintColor || palette.icon.primary, [props.tintColor, palette])

    if (!props.canGoBack) {
      return null
    }

    return (
      <TouchableOpacity onPress={props.onPress} activeOpacity={0.8} style={styles.container} ref={ref}>
        <Image source={iconsSources.closeCross} style={{ width: 24, height: 24 }} tintColor={tintColor} />
      </TouchableOpacity>
    )
  }),
)
HeaderCloseModalButton.displayName = 'HeaderCloseModalButton'

export { HeaderCloseModalButton }

const styles = StyleSheet.create({
  container: {
    // borderWidth: 1,
    // borderColor: 'red',
  },
})
