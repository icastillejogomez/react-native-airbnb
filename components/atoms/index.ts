export * from './HeaderCloseModalButton'
export * from './TabBarIcon'
export * from './TabBarLabel'
