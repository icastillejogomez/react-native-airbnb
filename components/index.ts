export * from './atoms'
export * from './molecules'
export * from './native'
export * from './organisms'
export * from './templates'
export * from './screens'
