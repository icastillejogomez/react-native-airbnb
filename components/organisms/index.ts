export * from './AirbnbTabBar'
export * from './AuthEmailForm'
export * from './AuthPhoneForm'
