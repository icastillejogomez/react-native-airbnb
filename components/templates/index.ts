export * from './AirbnbMainApplicationLayout'
export * from './AirbnbModalLayout'
