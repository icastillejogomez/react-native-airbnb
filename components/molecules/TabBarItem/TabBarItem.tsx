import { FC, ReactNode } from 'react'
import { StyleSheet, View } from 'react-native'
import { TabBarLabel } from '@/components/atoms'

export type TabBarItemProps = {
  icon: ReactNode
  isFocused: boolean
  label: string
}

const TabBarItem: FC<TabBarItemProps> = ({ icon, isFocused, label }) => {
  return (
    <View style={styles.container}>
      {icon}
      <TabBarLabel isFocused={isFocused} label={label} />
    </View>
  )
}
TabBarItem.displayName = 'TabBarItem'

export { TabBarItem }

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    gap: 4,
  },
})
