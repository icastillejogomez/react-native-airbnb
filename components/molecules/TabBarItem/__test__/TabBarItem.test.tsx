import { render } from '@testing-library/react-native'
import { TabBarItem } from '../TabBarItem'
import { TabBarIcon } from '@/components/atoms'

describe('TabBarItem', () => {
  it('should render', () => {
    render(<TabBarItem label={'Explore'} isFocused={true} icon={null} />)
  })

  it('should match snapshot', () => {
    expect(
      render(<TabBarItem label={'Explore'} isFocused={true} icon={<TabBarIcon tintColor={'red'} icon={'explore'} />} />),
    ).toMatchSnapshot()
  })
})
