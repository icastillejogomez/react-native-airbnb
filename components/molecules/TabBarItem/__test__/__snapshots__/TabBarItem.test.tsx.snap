// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarItem should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "gap": 4,
    }
  }
>
  <ViewManagerAdapter_ExpoImage
    cachePolicy="memory-disk"
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={26}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={
      [
        {
          "__packager_asset": true,
          "height": 1,
          "scale": 1,
          "uri": "file://assets/full/path/to/directory/name.type",
          "width": 1,
        },
      ]
    }
    style={
      {
        "height": 26,
        "width": 26,
      }
    }
    tintColor={4294901760}
    transition={null}
    width={26}
  />
  <RNCMaskedView>
    <View
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
            },
            {},
            {
              "fontFamily": "Montserrat-Medium",
              "fontSize": 12,
              "lineHeight": 20,
            },
            {
              "fontFamily": "Montserrat-SemiBold",
            },
            {},
            {},
            {
              "textAlign": "auto",
            },
            {
              "fontSize": 10,
            },
            {},
          ]
        }
      >
        Explore
      </Text>
    </View>
    <ViewManagerAdapter_ExpoLinearGradient
      colors={
        [
          4293271117,
          4293074015,
          4292281446,
        ]
      }
      endPoint={
        [
          1,
          0.5,
        ]
      }
      locations={
        [
          0,
          0.5,
          1,
        ]
      }
      startPoint={
        [
          0,
          0.5,
        ]
      }
    >
      <Text
        style={
          [
            [
              {
                "fontSize": 16,
              },
              {},
              {
                "fontFamily": "Montserrat-Medium",
                "fontSize": 12,
                "lineHeight": 20,
              },
              {
                "fontFamily": "Montserrat-SemiBold",
              },
              {},
              {},
              {
                "textAlign": "auto",
              },
              {
                "fontSize": 10,
              },
              {},
            ],
            {
              "opacity": 0,
            },
          ]
        }
      >
        Explore
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
  </RNCMaskedView>
</View>
`;
